sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/implementation/instrumentation/EventHubsTracer.java

```diff
@@ -148,74 +145,65 @@ private static boolean canModifyApplicationProperties(Map<String, Object> applic
         return !applicationProperties.getClass().getSimpleName().equals("UnmodifiableMap");
     }
 
-    public Context getBuilder(String spanName, Context context) {
-        return tracer == null ? context : setAttributes(tracer.getSharedSpanBuilder(spanName, context));
-    }
+    public TracingLink createLink(Map<String, Object> applicationProperties, Instant enqueuedTime, Context eventContext) {
+        Context link;
+        Optional<Object> linkContext = eventContext.getData(SPAN_CONTEXT_KEY);
+        if (linkContext.isPresent()) {
+            link = linkContext.get() instanceof Context ? (Context) linkContext.get() : Context.NONE;
```

Comment:
Should an informational or verbose message be logged here? This seems like an invalid state that can be recovered from